BuildResidentialHPXML/measure.rb

```diff
@@ -3318,20 +3345,63 @@ def self.set_air_infiltration_measurements(hpxml, runner, args)
   def self.set_attics(hpxml, runner, model, args)
     return if args[:geometry_unit_type] == HPXML::ResidentialTypeApartment
 
+    surf_ids = { 'roofs' => { 'surfaces' => hpxml.roofs, 'ids' => [] },
+                 'walls' => { 'surfaces' => hpxml.walls, 'ids' => [] },
+                 'frame_floors' => { 'surfaces' => hpxml.frame_floors, 'ids' => [] } }
+
+    attic_locations = [HPXML::LocationAtticUnconditioned, HPXML::LocationAtticUnvented, HPXML::LocationAtticVented]
+    surf_ids.each do |surf_type, surf_hash|
+      surf_hash['surfaces'].each do |surface|
+        next if (not attic_locations.include? surface.interior_adjacent_to) &&
+                (not attic_locations.include? surface.exterior_adjacent_to)
+
+        surf_hash['ids'] << surface.id
+      end
+    end
+
     if args[:geometry_roof_type] == 'flat'
       hpxml.attics.add(id: HPXML::AtticTypeFlatRoof,
                        attic_type: HPXML::AtticTypeFlatRoof)
     else
       hpxml.attics.add(id: args[:geometry_attic_type],
-                       attic_type: args[:geometry_attic_type])
+                       attic_type: args[:geometry_attic_type],
+                       attached_to_roof_idrefs: surf_ids['roofs']['ids'],
+                       attached_to_wall_idrefs: surf_ids['walls']['ids'],
+                       attached_to_frame_floor_idrefs: surf_ids['frame_floors']['ids'])
     end
   end
 
   def self.set_foundations(hpxml, runner, model, args)
     return if args[:geometry_unit_type] == HPXML::ResidentialTypeApartment
 
+    surf_ids = { 'slabs' => { 'surfaces' => hpxml.slabs, 'ids' => [] },
+                 'frame_floors' => { 'surfaces' => hpxml.frame_floors, 'ids' => [] },
+                 'foundation_walls' => { 'surfaces' => hpxml.foundation_walls, 'ids' => [] },
+                 'walls' => { 'surfaces' => hpxml.walls, 'ids' => [] },
+                 'rim_joists' => { 'surfaces' => hpxml.rim_joists, 'ids' => [] }, }
+
+    foundation_locations = [HPXML::LocationBasementConditioned, HPXML::LocationBasementUnconditioned,
+                            HPXML::LocationCrawlspaceUnvented, HPXML::LocationCrawlspaceVented,
+                            HPXML::FoundationTypeAmbient]
+
+    surf_ids.each do |surf_type, surf_hash|
+      surf_hash['surfaces'].each do |surface|
+        next if (not foundation_locations.include? surface.interior_adjacent_to) &&
+                (not foundation_locations.include? surface.exterior_adjacent_to) &&
+                (surf_type != 'slabs') &&
+                (surf_type != 'foundation_walls')
```

Comment:
Why is this code here? Doesn't that mean you won't get the surfaces adjacent to, e.g., a frame floor or rim joist?

Reply:
This ensures that we are always including slabs and foundation walls to surfaces attached to foundations, however I don't think `foundation_walls` is necessary, since those will always be adjacent to the foundation space which is already checked.

Reply:
I'm still confused. Slabs and foundation walls will have the same InteriorAdjacentTo; if it's not necessary for foundation walls, it shouldn't be necessary for slabs. Or am I missing something?

Reply:
The slabs check is only needed when there is a slab foundation, which is adjacent to the ground and the living space.

Reply:
As written, won't the logic end up including e.g. a garage slab as attached to the foundation?

Reply:
Yes it would, should it only include living space slabs?

Reply:
Yes

Reply:
Addressed in https://github.com/NREL/OpenStudio-HPXML/pull/820/commits/e8407f1b3579308e4ba42a95f5b69bc2e8631931 and re-wrote the logic to make it easier to understand